CYF London | Anna Fedyna | Module-Data-Flow | week 2 | book library fixed

annafedyna opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

cjyuan commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,103 @@
-    });
+    // Insert updated rows and cells
+    myLibrary.forEach((book, i) => {
+      let row = table.insertRow();
+      let cell1 = row.insertCell(0);
+      let cell2 = row.insertCell(1);
+      let cell3 = row.insertCell(2);
+      let cell4 = row.insertCell(3);
+      let cell5 = row.insertCell(4);
```
Note: These variables have been renamed in CYF's `main`.

annafedyna replied:
thanks, changed

cjyuan commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,103 @@
+      let changeBut = document.createElement("button");
+      changeBut.className = "btn btn-success";
+      cell4.appendChild(changeBut);
+      let readStatus = book.check ? "No" : "Yes"; 
```
**Todo**
Does `book.check == true` mean "Read" or "Not yet read"?

annafedyna replied:
yes, it does

cjyuan reviewed (commented):
1. There are few bugs (I marked them with "**Todo**" in the comments).
2. Some of the files and code are not used. It will make reviewing code easier if you can keep them in a separate folder or remove them from the branch.

SallyMcGrath commented on debugging/book-library/index.html:
```diff
@@ -65,7 +65,7 @@ <h1>Library</h1>
           value="Submit"
           class="btn btn-primary"
-          onclick="submit();"
+          data-atr = "submit"
```
Oh, btw, I generally advise data attributes, so perhaps because I've said so! 😉 

With data attributes you pass JSON, you can grab the entire dataset, and you can group like classes but not get in the way of CSS https://developer.mozilla.org/en-US/docs/Learn_web_development/Howto/Solve_HTML_problems/Use_data_attributes

annafedyna replied:
Thank you, it was easy to use as well

annafedyna commented on debugging/book-library/index.html:
```diff
@@ -79,18 +79,23 @@ <h1>Library</h1>
           <th></th>
         </tr>
+
+        <tbody id="book-list">
+        </tbody>
```
no, I removed it

annafedyna commented on debugging/book-library/index.html:
```diff
@@ -79,18 +79,23 @@ <h1>Library</h1>
-          <td></td>
-          <td></td>
-          <td></td>
+          <td data-title></td>
+          <td data-author></td>
+          <td data-pages></td>
+          <td data-read></td>
+          <td><button data-remove>Remove</button></td>
```
actually,they are not used, so I removed them as well

annafedyna commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,103 @@
+      );
+      myLibrary.push(book1);
+      myLibrary.push(book2);
+      render();
```
thank you, removed one

annafedyna commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,103 @@
+      pages.value === ""
+    ) {
+      alert("Please fill all fields!");
+      return false;
```
you right, that line is unnecessary, I removed it

annafedyna commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,103 @@
-    render();
+  function submit() {
+    if (
+      title.value == null ||
+      title.value === "" ||
+      pages.value == null ||
+      pages.value === ""
```
Thanks, you right, value can not be null. Also. I added more checks.

annafedyna commented:
> 1. There are few bugs (I marked them with "**Todo**" in the comments).
> 2. Some of the files and code are not used. It will make reviewing code easier if you can keep them in a separate folder or remove them from the branch.

Thank you for the review, appreciate it! Fixed bugs and removed unused code.

cjyuan commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,101 @@
-    render();
+  function submit() {
+    if (
+      title.value.trim() === "" ||
+      pages.value.trim() === "" ||
+      isNaN(pages.value) ||
+      parseInt(pages.value) <= 0 ||
+      !Number.isInteger(pages.value)
```
Still have some value not checked.

Some of these checks are redundant for ensuring `pages.value` represents a positive integer.
```
 pages.value.trim() === "" ||
      isNaN(pages.value) ||
      parseInt(pages.value) <= 0 ||
      !Number.isInteger(pages.value)
```

cjyuan commented on debugging/book-library/script.js:
```diff
@@ -1,103 +1,101 @@
+    ) {
+      alert("Please fill all fields appropriate!");
+    } else {
+      let book = new Book(
+        title.value,
+        author.value,
+        pages.value,
+        check.checked
+      );
```
Using input values directly without first sanitise them is dangerous.

What if the user enters "                                      foo" as title, or "000000007" as pages?

And if you have time for more challenges, what if the user enters 
"</table> haha" or "Hello &copy" as title?

cjyuan reviewed (commented):
Still have a bug.

Also, I added some more challenges for you.
